app/src/main/java/app/revanced/integrations/shared/GmsCoreSupport.java

```diff
@@ -40,48 +50,99 @@ private static void open(String queryOrLink, String message) {
         intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         Utils.getContext().startActivity(intent);
 
-        // Gracefully exit, otherwise without Gms the app crashes and Android can nag the user.
+        // Gracefully exit, otherwise the broken app will continue to run.
         System.exit(0);
     }
 
+    private static void showToastOrDialog(Context context, String toastMessageKey, String dialogMessageKey, String link) {
```

Comment:
Unfortunately, on WSA it seems the dialog is also not shown, and the app crashes if GmsCore is not installed. This might be due to GMS not being installed at all although GmsCore support patch changes most references to GmsCore. 

I have captured the following logs:

This one occurs from a check in the app. I have managed to circumvent it once by returning the correct enum integer value in one of the check methods (0), but the app may still crash:
```
03-30 03:23:56.809  1999  2144 E GooglePlayServicesUtil: GooglePlayServices not available due to error 9
03-30 03:23:56.809  1999  2144 E ttq     : Attempted to use SSL unpatched. Google Play Services unavailable.
03-30 03:23:56.809  1999  2144 E ttq     : ord
03-30 03:23:56.809  1999  2144 E ttq     :      at orf.d(PG:6)
```
```
03-30 03:19:36.874  1282  1389 E YouTube : [GservicesConfigHelper][preloadGservicesCacheBlocking] java.lang.SecurityException: Failed to find provider com.google.android.gsf.gservices for user 0; expected to find a valid ContentProvider for this authority
03-30 03:19:36.954  1282  1339 E ActivityThread: Failed to find provider info for com.google.android.gsf.gservices
03-30 03:19:37.070  1282  1452 E DynamiteModule: Invalid GmsCore APK, remote loading disabled.
03-30 03:19:37.071  1282  1452 E DynamiteModule: Invalid GmsCore APK, remote loading disabled.
03-30 03:19:37.073  1282  1459 E FirebaseInstanceId: Google Play services missing or without correct permission.
0
```